pkg/test/e2e/e2eutil/spec_util.go

```diff
@@ -33,14 +33,43 @@ func NewDefaultCluster(namespace string) *api.PravegaCluster {
 			Name:      "pravega",
 			Namespace: namespace,
 		},
-		Spec: api.ClusterSpec{},
+		Spec: api.ClusterSpec{
+			Pravega: &api.PravegaSpec{
+				Options:                map[string]string{"pravegaservice.cache.size.max": "1610612736"},
+				SegmentStoreJVMOptions: []string{"-Xmx1g", "-XX:MaxDirectMemorySize=2560m"},
+				SegmentStoreResources: &corev1.ResourceRequirements{
+					Requests: corev1.ResourceList{
+						corev1.ResourceCPU:    resource.MustParse("1000m"),
+						corev1.ResourceMemory: resource.MustParse("4Gi"),
+					},
+					Limits: corev1.ResourceList{
+						corev1.ResourceCPU:    resource.MustParse("2000m"),
+						corev1.ResourceMemory: resource.MustParse("4Gi"),
+					},
+				},
```

Comment:
Where do these values come from?

Reply:
Setting all the defaults here without verifying user supplied values is extremely dangerous.

If the user supplies memory limit less than `4Gi` (say, `2Gi`) without memory request,
they will end with a `4Gi` memory request (default) and `2Gi` memory limit (user-supplied).

What we rather need to do while setting the defaults is:
1. See if the user has set a memory request
2. If the request is not set, then set it to memory limit (the admission webhook would be verifying that the memory limit is set by the user).

Reply:
This default values are only for e2e tests and the checks are already in place for all the scenarios